include/inviwo/core/ports/datainport.h

```diff
@@ -214,12 +164,22 @@ void DataInport<T, N, Flat>::connectTo(Outport* port) {
 
 template <typename T, size_t N, bool Flat>
 bool DataInport<T, N, Flat>::isConnected() const {
-    return detail::isConnected<N>(connectedOutports_);
+    if constexpr (N == 0) {
+        return !connectedOutports_.empty();
+    } else {
+        return connectedOutports_.size() >= 1 && connectedOutports_.size() <= N;
+    }
 }
 
 template <typename T, size_t N, bool Flat>
 bool DataInport<T, N, Flat>::hasData() const {
-    return detail::HasData<T, N>::get(connectedOutports_, this->begin() != this->end());
+    if constexpr (N == 0) {
+        return isConnected() && this->begin() != this->end();
```

Comment:
Does the iterator internally check that the connected outports have data, i.e. `hasData() == true`.

Reply:
yes